packages/files/README.md

```diff
@@ -0,0 +1,193 @@
+### Files API
+
+This middleware has endpoints for uploading binaries and sharing previously uploaded files between apps/users.
+
+- **Base path**: `/api/files`
+- **Endpoints**:
+  - `POST /api/files/upload` — upload one or multiple files
+  - `POST /api/files/share` — create a new file record that points to an existing binary (re-share)
+
+### Authentication
+
+- Only authenticated, non-deleted users can call the endpoints. Auth is taken from the app session cookie (same as other Keystone-secured endpoints).
+- Unauthorized or deleted users receive `UNAUTHENTICATED/AUTHORIZATION_REQUIRED`.
+
+### Configuration
+
+Set `FILE_UPLOAD_CONFIG` env var to a JSON string:
+
+```json
+{
+  "clients": {
+    "some-app-internal-id": { "name": "Some app", "secret": "<HS256-signing-secret>" }
+  },
+  "quota": {
+    "user": 100,
+    "ip": 100
+  }
+}
+```
+
+### Uploading files
+
+- Method: `POST`
+- Content-Type: `multipart/form-data`
+- Fields:
+  - `file`: one or multiple file binaries
+  - `meta`: JSON string with:
+    - `dv`: `1`
+    - `sender`: `{ dv: 1, fingerprint: string }`
+    - `authedItemId`: UUID of the current user; must equal the authenticated user id
+    - `appId`: one of keys from `clients`
+    - `modelNames`: non-empty array of Models to which this file should be connected
+
+Example (curl):
+
+```bash
+curl -X POST "$SERVER/api/files/upload" \
+  -H "Cookie: $COOKIE" \
+  -F "file=@/path/to/dino.png" \
+  -F 'meta={"dv":1,"sender":{"dv":1,"fingerprint":"test-runner"},"authedItemId":"<USER_ID>","appId":"app-frontend","modelNames":["SomeModel"]}'
+```
+
+Successful response:
+
+```json
+{
+  "data": {
+    "files": [
+      { "id": "<uuid>", "signature": "<jwt>" }
+    ]
+  }
+}
+```
+
+- Multiple `file` parts are supported; response returns matching number of items.
+- `signature` is an HS256 JWT (expires in 5 minutes) signed with `clients[appId].secret`. Payload contains file meta, including `recordId`, original filename, mimetype, encoding, and `meta` with app/user info.
+
+### FileRecord model and signature payload
+
+The signed payload corresponds to the public meta of `FileRecord`:
+
+```json
+{
+  "id": "<storage-id>",
+  "recordId": "<FileRecord.id>",
+  "path": "<string|null>",
+  "filename": "<generated-name>",
+  "originalFilename": "<original name>",
+  "mimetype": "<mime>",
+  "encoding": "<encoding>",
+  "meta": {
+    "dv": 1,
+    "sender": { "dv": 1, "fingerprint": "<string>" },
+    "authedItemId": "<uuid>",
+    "appId": "<string>",
```

Comment:
I'm expecting something like: `user: { id: "<uuid>" }`, `app: { id: "<uuid>" }` and it's unclear which app ID should I use B2BApp OIDCApp B2CApp ... looks like we need to use some new word for that